### DISCUSSION TITLE
Tasmota BLE: How to ignore all unwanted BLE devices - whitelist

### DISCUSSION THREAD

**Discussion by ChrisVonFrosta** [arendst/Tasmota#15295] [Q&A] [answered]
Is there a way to whitlist the BLE devices?

I have a lot of Mi thermometers and Floras.
To cover the hole house i placed some ESP32 Boards with Tasmota BLE as BLE to MQTT Bridge.
Some Sensors are detected by several Tasmota BLE Bridges.
I would like to prevent this. I feel that behavior confusion for my use case.
There is no reason to get the same sensor values multiple per MQTT "channels".
Also if you move the Sensor and it get connection to other bridges, it will result in new MQTT Obejcts/Topics.

thx :)

**Comment by barbudor** [comment, marked as answer]
Check [Mi32Option5](https://tasmota.github.io/docs/Commands/#mi32option)

**Reply by BuxtonCalvin** [reply]
Hello,
This is one of the few discussions I've found for MI32Block.

Could you describe the list syntax for "MI32Block"  I've tried a number of ways-- such as the below --and can't get a list of macs to get accepted without an error.  I'm trying to create a rule to block all macs except one defined mac, from not just the UI but from processing as well.  The reason is that I have option Mi32Option2 1 to allow immediate transfer of BLE data to an MQTT message.  If I don't block the non essential macs from the UI and the console, it creates enormous MQTT traffic.

Oddly enough, I can add the below macs one at a time and they incrementally add to a block list.  But I can't form a rule this way (needed for persistence on reboots).  Using tasmota32c3-bluetooth 13.4.0.2 (2544cf0-bluetooth)

Thx

MI32Block A4C138DCD2AE A4C13898B338 A4C138A59E0C A4C13838EB2E A4C138957DDC A4C13878827C
MI32Block A4C138DCD2AE, A4C13898B338, A4C138A59E0C, A4C13838EB2E, A4C138957DDC, A4C13878827C
MI32Block=A4C138DCD2AE, A4C13898B338, A4C138A59E0C, A4C13838EB2E, A4C138957DDC, A4C13878827C

**Reply by sfromis** [reply]
The [docs clearly says](https://tasmota.github.io/docs/Commands/#ble-mi-sensors) that is to block **one** mac/alias, meaning that you need multiple commands for multiple blocks. I'd recommend doing this in `autoexec.bat` (one line per command) instead of messing around with rules.

**Reply by BuxtonCalvin** [reply]
I'll give an autoexec.bat a try, but the docs literally say:  "MI32Block = list blocked devices by mac".  I'm probably reading the description wrong in that MI32Block on its own will **return** a list, as opposed to accepting a list, but that is not very clear from the description.

**Reply by sfromis** [reply]
It literally says:
"MI32Block `<mac or blealias>` = Block one mac/alias."

**one**

**Comment by ChrisVonFrosta** [comment]
thx for the hint. did not notice this commands before. Will this help for my question?
This says: "x=5 - 0 -> show all relevant BLE sensors, 1 -> show only sensors with a BLEAlias"
If i set BLEAlias to one Sensor, i would do this for all. So i am back at the beginn. All Bridges find all BleDevices.
I find a blacklist option but this is the opposite of my need. 
[Mi32Block](https://tasmota.github.io/docs/Commands/#mi32block)

Maybe the whitlist could be a feature request?
The Blacklist (BlockOption) ist still there.

**Reply by sfromis** [reply]
Well, read again. With that option it should only report the sensors where you have assigned a BLEAlias name to the sensor. This is done separately for each Tasmota instance. That is the whitelist you asked for.

**Reply by ChrisVonFrosta** [reply]
expect the BLEAlias function under MIxy commands and found nothing. so i thougt BLEAlias means that name you can set while flashing the MI with ATC_Thermometer/Telink Flasher. My fault.
Now if found it with ctrl+F ;)
[BLEAlias](https://tasmota.github.io/docs/Commands/#ble-esp32)
thx a lot!

**Reply by ChrisVonFrosta** [reply]
I have to ask again.
Worked well - until restart.
After restart tasmota losts all alias?!

i have done a config backup before, but restore this did not help - did not restore the BLEAlias settings.

**Reply by sfromis** [reply]
You have to set the BLE aliases each boot, using autoexec.bat, a boot time rule, or whatever you prefer. This way, the potentially long list of names do not need to be stored in the limited-size area for settings.

**Comment by barbudor** [comment]
Indeed with ESP32 always having at least 4MB flash, a filesystem is always available.
You can have a autoexec.bat there that is a list of commands, one per line.
There is even an embedded file editor so you can create the file right in Tasmota

**Reply by ChrisVonFrosta** [reply]
after your hint i searched for examples and checked the docu.
But i am still unable to use the default commands like "BLEAlias MAC=AlIAS" In Berry Script.
I would try autoexec.be since i am on a esp32 but i found no examples for commands from the [command list](https://tasmota.github.io/docs/Commands/)

Maybe autoexec.bat works with the "normal" set of commands? i have to try.
But the Example shows only commands in brakets and i never see or use those bevore.
The Example:

```
; clr screen
[z]
; draw full screen picture - corona.rgb file must exist in UFS storage
[x0y0P/corona.rgb]
; define index color
[dc19:31000]
; draw transparent text with new index color
[x60y30f2Ci19D2]Tasmota
```

Maybe you like to give me a little example/hint again? ^^

**Reply by sfromis** [reply]
`autoexec.bat` is indeed just plain commands, one per line without any wrapping.
The example you posted is from `display.bat` which is a very different thing.
Commands from Berry is like `tasmota.cmd('BLEAlias MAC=AlIAS')`

**Reply by barbudor** [reply]
https://tasmota.github.io/docs/UFS/#autoexecbat

![image](https://user-images.githubusercontent.com/2996042/162193892-32da74bb-368f-4454-9d3e-2bbff4f1b2ef.png)

Looks pretty clear
Here is my own `autoexec.bat`
```
SetOption115 1
mi32option5 1
mi32option6 1
mi32page 10
blealias A4C138352C48=salon
blealias a4c138c7c998=garage
blealias a4c138be32d8=chbre_jmv
blealias a4c13860b9f8=salle_bain
blealias A4C1389D6A16=cuisine
blealias A4C13892D36D=chbre_coline
blealias A4C1382BDB8E=chbre_lucas
blealias A4C138C09AD3=chbre_manon
```

**Reply by ChrisVonFrosta** [reply]
THX a lot and a lot more ;)
